projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,603 @@
+    /** @hidden */
+    @ViewChild(IgxDateSingleComponent)
+    public single: IgxDateSingleComponent;
+
+    /** @hidden */
+    @ViewChild(IgxCalendarComponent)
+    public calendar: IgxCalendarComponent;
+
+    /** @hidden */
+    @ViewChild(IgxToggleDirective)
+    public toggleDirective: IgxToggleDirective;
+
+    /** @hidden */
+    @ContentChildren(IgxInputGroupBase)
+    public projectedInputs: QueryList<IgxInputGroupBase>;
+
+    /** @hidden */
+    @ContentChild(IgxDateRangePrefixDirective)
+    public prefix: IgxDateRangePrefixDirective;
+
+    /** @hidden */
+    @ContentChild(IgxDateRangeSuffixDirective)
+    public suffix: IgxDateRangeSuffixDirective;
+
+    @ContentChild(IgxLabelDirective)
+    public label: IgxLabelDirective;
+
+    /** @hidden */
+    public prepTo = CurrentResourceStrings.RangeDatePickerResStrings.igx_range_date_picker_to;
+
+    /** @hidden @internal */
+    public get appliedFormat() {
+        // TODO: apply formatter if present
+        if (!this.hasProjectedInputs) {
+            // TODO: use displayFormat - see how shortDate, longDate can be defined
+            return this.inputFormat
+                ? `${this.inputFormat} - ${this.inputFormat}`
+                : `${this._defaultInputFormat} - ${this._defaultInputFormat}`;
+        } else {
+            return this.inputFormat ? this.inputFormat : this._defaultInputFormat;
+        }
+    }
+
+    /** @hidden @internal */
+    public get hasProjectedInputs(): boolean {
+        return this.projectedInputs.some(i => i instanceof IgxDateStartComponent || i instanceof IgxDateEndComponent);
+    }
+
+    // TODO: use default input format from date-time-editor.common?
+    private _defaultInputFormat = 'dd/MM/yyyy';
+    private _value: DateRange;
+    private _destroy = new Subject<boolean>();
+    private _onChangeCallback: (_: any) => void;
+    private _positionSettings: PositionSettings;
+    private _positionStrategy: AutoPositionStrategy;
+    private _dialogOverlaySettings: OverlaySettings;
+    private _dropDownOverlaySettings: OverlaySettings;
+
+    constructor(public element: ElementRef) {
+        this._onChangeCallback = (_: any) => { };
+    }
+
+    /**
+     * Opens the date range picker's dropdown or dialog.
+     *
+     * @example
+     * ```html
+     * <igx-date-range #dateRange mode="dialog"></igx-date-range>
+     *
+     * <button (click)="dateRange.open()">Open Dialog</button
+     * ```
+     */
+    public open(): void {
+        if (this.value && this.value.start?.getTime() > this.value.end?.getTime()) {
+            this.value = { start: this.value.end, end: this.value.start };
+        }
+        this.updateCalendar();
+        if (this.mode === InteractionMode.Dialog) {
+            this.activateToggleOpen(this._dialogOverlaySettings);
+        }
+        if (this.mode === InteractionMode.DropDown) {
+            this.activateToggleOpen(this._dropDownOverlaySettings);
+        }
+
+        this.onOpened.emit({ owner: this });
```

Comment:
this should be called only when it is opened in `this.toggleDirective.onOpened`
you can call it and in `activateToggleOpen` after the call to `open` but this will not reflect the real end of opening